src/components/TodoForm.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import { useTodoContext } from '../context/TodoContext';
+import { addTodo } from '../api/todos';
+import { TodoError } from '../types/Error';
+
+export const TodoForm: React.FC = () => {
+  const {
+    setTodos,
+    setError,
+  } = useTodoContext();
+
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isCreating, setIsCreating] = useState(false);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setError(TodoError.INVALID_INPUT);
+
+      return;
+    }
+
+    setError(null);
+    setIsCreating(true);
+
+    const newTodo = {
+      id: 0,
```

Comment:
hardcoded id not the best idea
I think you no need this param in this case at all
```suggestion
```